gammapy/maps/wcsnd.py

```diff
@@ -495,38 +498,116 @@ def plot(self, ax=None, fig=None, add_cbar=False, stretch='linear', **kwargs):
-        except AttributeError:
-            log.info("Can't set coordinate axes. No WCS information available.")
-
-        # without this the axis limits are changed when calling scatter
-        ax.autoscale(enable=False)
-        return fig, ax, cbar
+
+        if self.geom.ndim == 2:
+            if ax is None:
+                ax = fig.add_subplot(1, 1, 1, projection=self.geom.wcs)
+
+            data = self.data
+            norm = simple_norm(data[np.isfinite(data)], stretch)
+            kwargs.setdefault('norm', norm)
+            caxes = ax.imshow(data, **kwargs)
+            cbar = fig.colorbar(caxes, ax=ax) if add_cbar else None
+            # without this the axis limits are changed when calling scatter
+            ax.autoscale(enable=False)
+
+            try:
+                ax.coords['glon'].set_axislabel('Galactic Longitude')
+                ax.coords['glat'].set_axislabel('Galactic Latitude')
+            except KeyError:
+                ax.coords['ra'].set_axislabel('Right Ascension')
+                ax.coords['dec'].set_axislabel('Declination')
+            except AttributeError:
+                log.info("Can't set coordinate axes. No WCS information available.")
+
+            return fig, ax, cbar
+
+        elif self.geom.ndim == 3:
+
+            try:
+                from ipywidgets.widgets.interaction import interact, fixed
+                import ipywidgets as widgets
+            except:
+                raise ImportError('ipywidgets is not installed.')
+
+            @interact(
+                     index=widgets.IntSlider(min=0, max=self.data.shape[0] - 1, step=1, value=1,
+                                             description=self.geom.axes_names[0]+' slice'),
+                     stretch=widgets.RadioButtons(options=['linear', 'sqrt', 'log'], value='sqrt',
+                                                  description='Plot stretch'),
+                     mapND=fixed(self)
+                    )
+            def _plot_interactive(mapND, index, stretch='linear'):
+                """
+                Plot ND array on matplotlib WCS axes with interactive widgets
+
+                index : integer
+                    Slice index of the non spatial dimension.
+                stretch : str
+                    Passed to `astropy.visualization.simple_norm`
+                Returns
+                -------
+                #TODO: currently returns nothing.
+                        Not sure if interact can return something.
+                        Was producing errors when I tried.
+
+                fig : `~matplotlib.figure.Figure`
+                    Figure object.
+                ax : `~astropy.visualization.wcsaxes.WCSAxes`
+                    WCS axis object
+                cbar : `~matplotlib.colorbar.Colorbar` or None
+                    Colorbar object.
+                """
+
+                fig = plt.gcf()
+
+                fig.set_size_inches(12, 5)
```

Comment:
Is it possible to remove this line, because it prevents the user from setting the figure size from outside. E.g. by:
```python
plt.figure(figsize=(16, 8))
map_nd.plot()
```

Reply:
I think this should be easy to fix.

Reply:
So turns out that this is not an easy fix as for some reason the fig = plt.gcf() needs to be created inside the function. When I tried to pass it as an argument no plot is shown.

Reply:
The call to `fig = plt.gcf()` is OK, I just meant hard-coding the size of the figure in line 574, as this prevents the user from setting it from outside. E.g. I like to adapt the figure size, when plotting survey images to make them appear larger on the screen.

Reply:
If I don't specify anything I get a ~square image like that:

![image](https://user-images.githubusercontent.com/8652898/42940224-f85069ca-8b58-11e8-9259-3f33cada53fb.png)
When nothing is specified, giving this `plt.figure(figsize=(16, 8))`before the .plot() doesn't change the size ratio. So not sure how to handle this.

Reply:
Maybe you could try the same using the mpl rc context manager:
```python
import matplotlib as mpl
with mpl.rc_context(rc={'figure.figsize': (16, 8)}):
    wcs_nd.plot()
```
If you think this is to complex, then maybe add an optional `figsize` argument to the `plot_interactive()` method?